### ISSUE TITLE
Refine tracklets Attributeerror

### ISSUE THREAD

**Issue by woohyunk1104** [DeepLabCut/DeepLabCut#1671]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Bug description

While refining tracklets, when I click on individual 1 to swap the labels for individual 1 and 2, I get an attributeerror in the terminal. 

![Figure_1](https://user-images.githubusercontent.com/47169385/152520014-8e0d9e76-78ef-40e3-93b7-6a00245c25ba.png)
1

(base) csbd@csbd-System-Product-Name:~$ conda activate DEEPLABCUT
(DEEPLABCUT) csbd@csbd-System-Product-Name:~$ python -m deeplabcut
/home/csbd/anaconda3/envs/DEEPLABCUT/lib/python3.8/site-packages/statsmodels/compat/pandas.py:65: FutureWarning: pandas.Int64Index is deprecated and will be removed from pandas in a future version. Use pandas.Index with the appropriate dtype instead.
  from pandas import Int64Index as NumericIndex
Starting GUI...
Traceback (most recent call last):
  File "/home/csbd/anaconda3/envs/DEEPLABCUT/lib/python3.8/site-packages/matplotlib/cbook/__init__.py", line 287, in process
    func(*args, **kwargs)
  File "/home/csbd/anaconda3/envs/DEEPLABCUT/lib/python3.8/site-packages/deeplabcut/gui/tracklet_toolbox.py", line 511, in on_pick
    self.ax_slider.lines = []
AttributeError: can't set attribute

### Operating System

ubuntu20.04

### DeepLabCut version

dlc version latest version (2.0.6)

### DeepLabCut mode

multi animal

### Device type

I want to use the gpu (gpu 3090), but seems like I'm using the cpu.

### Steps To Reproduce

(DEEPLABCUT) csbd@csbd-System-Product-Name:~$ conda list

### Relevant log output

```shell
# packages in environment at /home/csbd/anaconda3/envs/DEEPLABCUT:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       1_gnu    conda-forge
absl-py                   1.0.0                    pypi_0    pypi
alsa-lib                  1.2.3                h516909a_0    conda-forge
aom                       3.2.0                h9c3ff4c_2    conda-forge
argon2-cffi               21.3.0             pyhd8ed1ab_0    conda-forge
argon2-cffi-bindings      21.2.0           py38h497a2fe_1    conda-forge
asttokens                 2.0.5              pyhd8ed1ab_0    conda-forge
astunparse                1.6.3                    pypi_0    pypi
attrs                     21.4.0             pyhd8ed1ab_0    conda-forge
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
backports                 1.0                        py_2    conda-forge
backports.functools_lru_cache 1.6.4              pyhd8ed1ab_0    conda-forge
black                     22.1.0             pyhd8ed1ab_0    conda-forge
bleach                    4.1.0              pyhd8ed1ab_0    conda-forge
bzip2                     1.0.8                h7f98852_4    conda-forge
ca-certificates           2021.10.8            ha878542_0    conda-forge
cachetools                5.0.0                    pypi_0    pypi
certifi                   2021.10.8                pypi_0    pypi
cffi                      1.15.0           py38h3931269_0    conda-forge
charset-normalizer        2.0.11                   pypi_0    pypi
click                     8.0.3            py38h578d9bd_1    conda-forge
cycler                    0.11.0                   pypi_0    pypi
dataclasses               0.8                pyhc8e2a94_3    conda-forge
dbus                      1.13.6               h5008d03_3    conda-forge
debugpy                   1.5.1            py38h709712a_0    conda-forge
decorator                 4.4.2                    pypi_0    pypi
deeplabcut                2.2.0.6                  pypi_0    pypi
defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
entrypoints               0.4                pyhd8ed1ab_0    conda-forge
executing                 0.8.2              pyhd8ed1ab_0    conda-forge
expat                     2.4.4                h9c3ff4c_0    conda-forge
ffmpeg                    4.4.1                h6987444_0    conda-forge
filterpy                  1.4.5                    pypi_0    pypi
flatbuffers               2.0                      pypi_0    pypi
flit-core                 3.6.0              pyhd8ed1ab_0    conda-forge
font-ttf-dejavu-sans-mono 2.37                 hab24e00_0    conda-forge
font-ttf-inconsolata      3.000                h77eed37_0    conda-forge
font-ttf-source-code-pro  2.038                h77eed37_0    conda-forge
font-ttf-ubuntu           0.83                 hab24e00_0    conda-forge
fontconfig                2.13.94              ha180cfb_0    conda-forge
fonts-conda-ecosystem     1                             0    conda-forge
fonts-conda-forge         1                             0    conda-forge
fonttools                 4.29.1                   pypi_0    pypi
freetype                  2.10.4               h0708190_1    conda-forge
gast                      0.5.3                    pypi_0    pypi
gettext                   0.19.8.1          h73d1719_1008    conda-forge
gmp                       6.2.1                h58526e2_0    conda-forge
gnutls                    3.6.13               h85f3911_1    conda-forge
google-auth               2.6.0                    pypi_0    pypi
google-auth-oauthlib      0.4.6                    pypi_0    pypi
google-pasta              0.2.0                    pypi_0    pypi
grpcio                    1.43.0                   pypi_0    pypi
gst-plugins-base          1.18.5               hf529b03_3    conda-forge
gstreamer                 1.18.5               h9f60fe5_3    conda-forge
h5py                      3.6.0                    pypi_0    pypi
icu                       69.1                 h9c3ff4c_0    conda-forge
idna                      3.3                      pypi_0    pypi
imageio                   2.14.1                   pypi_0    pypi
imageio-ffmpeg            0.4.5                    pypi_0    pypi
imgaug                    0.4.0                    pypi_0    pypi
importlib-metadata        4.10.1           py38h578d9bd_0    conda-forge
importlib_resources       5.4.0              pyhd8ed1ab_0    conda-forge
ipykernel                 6.8.0            py38he5a9106_0    conda-forge
ipython                   8.0.1            py38h578d9bd_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
ipywidgets                7.6.5              pyhd8ed1ab_0    conda-forge
jedi                      0.18.1           py38h578d9bd_0    conda-forge
jinja2                    3.0.3              pyhd8ed1ab_0    conda-forge
joblib                    1.1.0                    pypi_0    pypi
jpeg                      9e                   h7f98852_0    conda-forge
jsonschema                4.4.0              pyhd8ed1ab_0    conda-forge
jupyter                   1.0.0            py38h578d9bd_7    conda-forge
jupyter_client            7.1.2              pyhd8ed1ab_0    conda-forge
jupyter_console           6.4.0              pyhd8ed1ab_0    conda-forge
jupyter_core              4.9.1            py38h578d9bd_1    conda-forge
jupyterlab_pygments       0.1.2              pyh9f0ad1d_0    conda-forge
jupyterlab_widgets        1.0.2              pyhd8ed1ab_0    conda-forge
keras                     2.8.0                    pypi_0    pypi
keras-preprocessing       1.1.2                    pypi_0    pypi
kiwisolver                1.3.2                    pypi_0    pypi
krb5                      1.19.2               hcc1bbae_3    conda-forge
lame                      3.100             h7f98852_1001    conda-forge
ld_impl_linux-64          2.36.1               hea4e1c9_2    conda-forge
libclang                  13.0.0                   pypi_0    pypi
libdrm                    2.4.109              h7f98852_0    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libevent                  2.1.10               h9b69904_4    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 11.2.0              h1d223b6_12    conda-forge
libglib                   2.70.2               h174f98d_1    conda-forge
libgomp                   11.2.0              h1d223b6_12    conda-forge
libiconv                  1.16                 h516909a_0    conda-forge
libllvm13                 13.0.0               hf817b99_0    conda-forge
libnsl                    2.0.0                h7f98852_0    conda-forge
libogg                    1.3.4                h7f98852_1    conda-forge
libopus                   1.3.1                h7f98852_1    conda-forge
libpciaccess              0.16                 h516909a_0    conda-forge
libpng                    1.6.37               h21135ba_2    conda-forge
libpq                     14.1                 hd57d9b9_1    conda-forge
libsodium                 1.0.18               h36c2ea0_1    conda-forge
libstdcxx-ng              11.2.0              he4da1e4_12    conda-forge
libuuid                   2.32.1            h7f98852_1000    conda-forge
libva                     2.13.0               h7f98852_2    conda-forge
libvorbis                 1.3.7                h9c3ff4c_0    conda-forge
libvpx                    1.11.0               h9c3ff4c_3    conda-forge
libxcb                    1.13              h7f98852_1004    conda-forge
libxkbcommon              1.0.3                he3ba5ed_0    conda-forge
libxml2                   2.9.12               h885dcf4_1    conda-forge
libzlib                   1.2.11            h36c2ea0_1013    conda-forge
llvmlite                  0.38.0                   pypi_0    pypi
lz4-c                     1.9.3                h9c3ff4c_1    conda-forge
markdown                  3.3.6                    pypi_0    pypi
markupsafe                2.0.1            py38h497a2fe_1    conda-forge
matplotlib                3.5.1                    pypi_0    pypi
matplotlib-inline         0.1.3              pyhd8ed1ab_0    conda-forge
mistune                   0.8.4           py38h497a2fe_1005    conda-forge
moviepy                   1.0.3                    pypi_0    pypi
msgpack                   1.0.3                    pypi_0    pypi
msgpack-numpy             0.4.7.1                  pypi_0    pypi
mypy_extensions           0.4.3            py38h578d9bd_4    conda-forge
mysql-common              8.0.28               ha770c72_0    conda-forge
mysql-libs                8.0.28               hfa10184_0    conda-forge
nb_conda                  2.2.1            py38h578d9bd_4    conda-forge
nb_conda_kernels          2.3.1            py38h578d9bd_1    conda-forge
nbclient                  0.5.10             pyhd8ed1ab_1    conda-forge
nbconvert                 6.4.1            py38h578d9bd_0    conda-forge
nbformat                  5.1.3              pyhd8ed1ab_0    conda-forge
ncurses                   6.3                  h9c3ff4c_0    conda-forge
nest-asyncio              1.5.4              pyhd8ed1ab_0    conda-forge
nettle                    3.6                  he412f7d_0    conda-forge
networkx                  2.6.3                    pypi_0    pypi
notebook                  6.4.8              pyha770c72_0    conda-forge
nspr                      4.32                 h9c3ff4c_1    conda-forge
nss                       3.74                 hb5efdd6_0    conda-forge
numba                     0.55.1                   pypi_0    pypi
numexpr                   2.8.1                    pypi_0    pypi
numpy                     1.21.5                   pypi_0    pypi
oauthlib                  3.2.0                    pypi_0    pypi
opencv-python             4.5.5.62                 pypi_0    pypi
openh264                  2.1.1                h780b84a_0    conda-forge
openssl                   1.1.1l               h7f98852_0    conda-forge
opt-einsum                3.3.0                    pypi_0    pypi
packaging                 21.3               pyhd8ed1ab_0    conda-forge
pandas                    1.4.0                    pypi_0    pypi
pandoc                    2.17.1.1             ha770c72_0    conda-forge
pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
parso                     0.8.3              pyhd8ed1ab_0    conda-forge
pathspec                  0.9.0              pyhd8ed1ab_0    conda-forge
patsy                     0.5.2                    pypi_0    pypi
pcre                      8.45                 h9c3ff4c_0    conda-forge
pexpect                   4.8.0              pyh9f0ad1d_2    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pillow                    9.0.1                    pypi_0    pypi
pip                       22.0.3             pyhd8ed1ab_0    conda-forge
platformdirs              2.3.0              pyhd8ed1ab_0    conda-forge
proglog                   0.1.9                    pypi_0    pypi
prometheus_client         0.13.1             pyhd8ed1ab_0    conda-forge
prompt-toolkit            3.0.26             pyha770c72_0    conda-forge
prompt_toolkit            3.0.26               hd8ed1ab_0    conda-forge
protobuf                  3.19.4                   pypi_0    pypi
psutil                    5.9.0                    pypi_0    pypi
pthread-stubs             0.4               h36c2ea0_1001    conda-forge
ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
pure_eval                 0.2.2              pyhd8ed1ab_0    conda-forge
pyasn1                    0.4.8                    pypi_0    pypi
pyasn1-modules            0.2.8                    pypi_0    pypi
pycparser                 2.21               pyhd8ed1ab_0    conda-forge
pygments                  2.11.2             pyhd8ed1ab_0    conda-forge
pyparsing                 3.0.7              pyhd8ed1ab_0    conda-forge
pyqt                      5.12.3           py38h578d9bd_8    conda-forge
pyqt-impl                 5.12.3           py38h0ffb2e6_8    conda-forge
pyqt5-sip                 4.19.18          py38h709712a_8    conda-forge
pyqtchart                 5.12             py38h7400c14_8    conda-forge
pyqtwebengine             5.12.1           py38h7400c14_8    conda-forge
pyrsistent                0.18.1           py38h497a2fe_0    conda-forge
python                    3.8.12          ha38a3c6_3_cpython    conda-forge
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python_abi                3.8                      2_cp38    conda-forge
pytz                      2021.3                   pypi_0    pypi
pywavelets                1.2.0                    pypi_0    pypi
pyyaml                    6.0                      pypi_0    pypi
pyzmq                     22.3.0           py38h2035c66_1    conda-forge
qt                        5.12.9               ha98a1a1_5    conda-forge
qtconsole                 5.2.2              pyhd8ed1ab_1    conda-forge
qtconsole-base            5.2.2              pyhd8ed1ab_1    conda-forge
qtpy                      2.0.1              pyhd8ed1ab_0    conda-forge
readline                  8.1                  h46c0cb4_0    conda-forge
requests                  2.27.1                   pypi_0    pypi
requests-oauthlib         1.3.1                    pypi_0    pypi
rsa                       4.8                      pypi_0    pypi
ruamel-yaml               0.17.20                  pypi_0    pypi
ruamel-yaml-clib          0.2.6                    pypi_0    pypi
scikit-image              0.18.1                   pypi_0    pypi
scikit-learn              1.0.2                    pypi_0    pypi
scipy                     1.7.3                    pypi_0    pypi
send2trash                1.8.0              pyhd8ed1ab_0    conda-forge
setuptools                60.7.1           py38h578d9bd_0    conda-forge
shapely                   1.8.0                    pypi_0    pypi
six                       1.16.0             pyh6c4a22f_0    conda-forge
sqlite                    3.37.0               h9cd32fc_0    conda-forge
stack_data                0.1.4              pyhd8ed1ab_0    conda-forge
statsmodels               0.13.1                   pypi_0    pypi
svt-av1                   0.9.0                h9c3ff4c_0    conda-forge
tables                    3.6.1                    pypi_0    pypi
tabulate                  0.8.9                    pypi_0    pypi
tensorboard               2.8.0                    pypi_0    pypi
tensorboard-data-server   0.6.1                    pypi_0    pypi
tensorboard-plugin-wit    1.8.1                    pypi_0    pypi
tensorflow                2.8.0                    pypi_0    pypi
tensorflow-gpu            2.8.0                    pypi_0    pypi
tensorflow-io-gcs-filesystem 0.23.1                   pypi_0    pypi
tensorpack                0.11                     pypi_0    pypi
termcolor                 1.1.0                    pypi_0    pypi
terminado                 0.13.1           py38h578d9bd_0    conda-forge
testpath                  0.5.0              pyhd8ed1ab_0    conda-forge
tf-estimator-nightly      2.8.0.dev2021122109          pypi_0    pypi
tf-slim                   1.1.0                    pypi_0    pypi
threadpoolctl             3.1.0                    pypi_0    pypi
tifffile                  2022.2.2                 pypi_0    pypi
tk                        8.6.11               h27826a3_1    conda-forge
tomli                     2.0.0              pyhd8ed1ab_1    conda-forge
tornado                   6.1              py38h497a2fe_2    conda-forge
tqdm                      4.62.3                   pypi_0    pypi
traitlets                 5.1.1              pyhd8ed1ab_0    conda-forge
typed-ast                 1.5.2            py38h497a2fe_0    conda-forge
typing_extensions         4.0.1              pyha770c72_0    conda-forge
urllib3                   1.26.8                   pypi_0    pypi
wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
webencodings              0.5.1                      py_1    conda-forge
werkzeug                  2.0.2                    pypi_0    pypi
wheel                     0.37.1             pyhd8ed1ab_0    conda-forge
widgetsnbextension        3.5.2            py38h578d9bd_1    conda-forge
wrapt                     1.13.3                   pypi_0    pypi
wxpython                  4.1.1                    pypi_0    pypi
x264                      1!161.3030           h7f98852_1    conda-forge
x265                      3.5                  h4bd325d_1    conda-forge
xorg-fixesproto           5.0               h7f98852_1002    conda-forge
xorg-kbproto              1.0.7             h7f98852_1002    conda-forge
xorg-libx11               1.7.2                h7f98852_0    conda-forge
xorg-libxau               1.0.9                h7f98852_0    conda-forge
xorg-libxdmcp             1.1.3                h7f98852_0    conda-forge
xorg-libxext              1.3.4                h7f98852_1    conda-forge
xorg-libxfixes            5.0.3             h7f98852_1004    conda-forge
xorg-xextproto            7.3.0             h7f98852_1002    conda-forge
xorg-xproto               7.0.31            h7f98852_1007    conda-forge
xz                        5.2.5                h516909a_1    conda-forge
zeromq                    4.3.4                h9c3ff4c_1    conda-forge
zipp                      3.7.0              pyhd8ed1ab_1    conda-forge
zlib                      1.2.11            h36c2ea0_1013    conda-forge
zstd                      1.5.2                ha95c52a_0    conda-forge
```

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/DeepLabCut/DeepLabCut/blob/master/CODE_OF_CONDUCT.md)

**Comment by jeylau** [comment]
Could you try to downgrade matplotlib with `pip install matplotlib==3.5.0` and tell me if that fixes your error?

**Comment by woohyunk1104** [comment]
> Could you try to downgrade matplotlib with `pip install matplotlib==3.5.0` and tell me if that fixes your error?

Thank you for your quick reply.
I tried downgrading, but it doesn't fix my error. 

(DEEPLABCUT) csbd@csbd-System-Product-Name:~$ pip install matplotlib==3.5.0
Collecting matplotlib==3.5.0
  Using cached matplotlib-3.5.0-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.whl (11.3 MB)
Requirement already satisfied: fonttools>=4.22.0 in ./anaconda3/envs/DEEPLABCUT/lib/python3.8/site-packages (from matplotlib==3.5.0) (4.29.1)
Requirement already satisfied: pyparsing>=2.2.1 in ./anaconda3/envs/DEEPLABCUT/lib/python3.8/site-packages (from matplotlib==3.5.0) (3.0.7)
Collecting setuptools-scm>=4
  Using cached setuptools_scm-6.4.2-py3-none-any.whl (37 kB)
Requirement already satisfied: numpy>=1.17 in ./anaconda3/envs/DEEPLABCUT/lib/python3.8/site-packages (from matplotlib==3.5.0) (1.21.5)
Requirement already satisfied: kiwisolver>=1.0.1 in ./anaconda3/envs/DEEPLABCUT/lib/python3.8/site-packages (from matplotlib==3.5.0) (1.3.2)
Requirement already satisfied: python-dateutil>=2.7 in ./anaconda3/envs/DEEPLABCUT/lib/python3.8/site-packages (from matplotlib==3.5.0) (2.8.2)
Requirement already satisfied: pillow>=6.2.0 in ./anaconda3/envs/DEEPLABCUT/lib/python3.8/site-packages (from matplotlib==3.5.0) (9.0.1)
Requirement already satisfied: packaging>=20.0 in ./anaconda3/envs/DEEPLABCUT/lib/python3.8/site-packages (from matplotlib==3.5.0) (21.3)
Requirement already satisfied: cycler>=0.10 in ./anaconda3/envs/DEEPLABCUT/lib/python3.8/site-packages (from matplotlib==3.5.0) (0.11.0)
Requirement already satisfied: six>=1.5 in ./anaconda3/envs/DEEPLABCUT/lib/python3.8/site-packages (from python-dateutil>=2.7->matplotlib==3.5.0) (1.16.0)
Requirement already satisfied: tomli>=1.0.0 in ./anaconda3/envs/DEEPLABCUT/lib/python3.8/site-packages (from setuptools-scm>=4->matplotlib==3.5.0) (2.0.0)
Requirement already satisfied: setuptools in ./anaconda3/envs/DEEPLABCUT/lib/python3.8/site-packages (from setuptools-scm>=4->matplotlib==3.5.0) (60.7.1)
Installing collected packages: setuptools-scm, matplotlib
  Attempting uninstall: matplotlib
    Found existing installation: matplotlib 3.5.1
    Uninstalling matplotlib-3.5.1:
      Successfully uninstalled matplotlib-3.5.1
Successfully installed matplotlib-3.5.0 setuptools-scm-6.4.2
(DEEPLABCUT) csbd@csbd-System-Product-Name:~$ pip show matplotlib
Name: matplotlib
Version: 3.5.0
Summary: Python plotting package
Home-page: https://matplotlib.org
Author: John D. Hunter, Michael Droettboom
Author-email: matplotlib-users@python.org
License: PSF
Location: /home/csbd/anaconda3/envs/DEEPLABCUT/lib/python3.8/site-packages
Requires: cycler, fonttools, kiwisolver, numpy, packaging, pillow, pyparsing, python-dateutil, setuptools-scm
Required-by: deeplabcut, filterpy, imgaug, scikit-image
(DEEPLABCUT) csbd@csbd-System-Product-Name:~$ python -m deeplabcut
/home/csbd/anaconda3/envs/DEEPLABCUT/lib/python3.8/site-packages/statsmodels/compat/pandas.py:65: FutureWarning: pandas.Int64Index is deprecated and will be removed from pandas in a future version. Use pandas.Index with the appropriate dtype instead.
  from pandas import Int64Index as NumericIndex
Starting GUI...
Traceback (most recent call last):
  File "/home/csbd/anaconda3/envs/DEEPLABCUT/lib/python3.8/site-packages/matplotlib/cbook/__init__.py", line 287, in process
    func(*args, **kwargs)
  File "/home/csbd/anaconda3/envs/DEEPLABCUT/lib/python3.8/site-packages/deeplabcut/gui/tracklet_toolbox.py", line 511, in on_pick
    self.ax_slider.lines = []
AttributeError: can't set attribute

**Comment by jeylau** [comment]
@woohyunk1104, I fixed your bug with the PR #1680. matplotlib introduced the change in version 3.5.0, so anything 3.4.x would work meanwhile, sorry about that 😃

**Comment by woohyunk1104** [comment]
> @woohyunk1104, I fixed your bug with the PR #1680. matplotlib introduced the change in version 3.5.0, so anything 3.4.x would work meanwhile, sorry about that 😃

Installing version 3.4.1 helped me solve the problem! Thank you!
